src/Models/ShallowWaterModels/shallow_water_model.jl

```diff
@@ -35,11 +35,11 @@ function ShallowWaterSolutionFields(grid, bcs, prognostic_names)
     return NamedTuple{prognostic_names[1:3]}((u, v, h))
 end
 
-mutable struct ShallowWaterModel{G, A<:AbstractArchitecture, T, V, U, R, F, E, B, Q, C, K, TS, FR} <: AbstractModel{TS}
+mutable struct ShallowWaterModel{G, A<:AbstractArchitecture, T, GR, V, U, R, F, E, B, Q, C, K, TS, FR} <: AbstractModel{TS}
                           grid :: G         # Grid of physical points on which `Model` is solved
                   architecture :: A         # Computer `Architecture` on which `Model` is run
                          clock :: Clock{T}  # Tracks iteration number and simulation time of `Model`
-    gravitational_acceleration :: T         # Gravitational acceleration, full, or reduced
+    gravitational_acceleration :: GR        # Gravitational acceleration, full, or reduced
```

Comment:
```suggestion
    gravitational_acceleration              # Gravitational acceleration, full, or reduced
```

Reply:
i disagree with this suggestion

Reply:
OK, I see. Better to have an extra parameter rather that having the danger of inducing a type instability